lists-and-memory/sorting-algorithms-lists-memory.md

```diff
@@ -0,0 +1,124 @@
+| ----------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Sorting                                   | Organizing a collection of elements into order                                                                                                                                      | "By sorting my list, I can present the data in a more understandable fashion."                                                                                                                                                           |
+| In-place Sorting                          | Sorting an array using only a small, constant amount—O(1)—of extra storage space. Sorting in place modifies the array to be sorted directly rather than making a new, sorted array. | "Our system is memory-constrained, so we should use an in-place sorting strategy so we don't have to duplicate our data." "Copy the array before using that in-place sort. We don't want to modify our original data in this situation." |
+| Stable                                    | A sorting algorithm is _stable_ if it guarantees that the relative ordering of two values having the same sort key is the same after the sort as it was before the sort.            | "Since that sorting algorithm is stable, we can apply it multiple times to the same data to sort records based on multiple attributes!"                                                                                                  |
+
+## Introduction
+
+Imagine a stack of index cards, where each index card has the numbers 1 through 100. Imagine that these cards are shuffled in a random, unknowable order.
+
+If you're given the task to sort these index cards in ascending order, what would you do?
+
+Picture yourself. Do you scan through the pile over and over again, slowly changing the order until it's all sorted? Do you use a table and make two separate piles? Do you fan through the cards comparing each card to its neighbor in the pile?
+
+If you take two people and ask them to sort the same pile of cards, they will likely take different approaches.
+
+Machines need to be told how to think through these questions too, but with fewer index cards and more data. The strategies we tell a computer to use for sorting a list of values are called _sorting algorithms_.
```

Comment:
```suggestion
Machines need to be told how to think through these questions too, but data instead of index cards. The strategies we tell a computer to use for sorting a list of values are called _sorting algorithms_.
```

Reply:
+1